Sources/PowerAuth/PowerAuthError.swift

```diff
@@ -0,0 +1,202 @@
+        case notSupported
+        /// The biometric authentication is not available on the device.
+        case notAvailable
+        /// There's no enrolled biometry on the device.
+        case notEnrolled
+        /// The biometric factor is not configured in `PowerAuth` instance.
+        case notConfigured
+    }
+    
+    /// User did cancel the biometric authentication dialog.
+    case biometricAuthenticationCancel
+    
+    /// Authentication with biometry failed. Check the reason for more details.
+    case biometricAuthenticationFailed(reason: BiometricFailureReason)
+    
+    /// The requested token doesn't exist.
+    case tokenNotFound
+    
+    
+    // Protocol upgrade
+    
+    /// The requested operation failed due to pending protocol upgrade.
+    /// You can retry the operation later.
+    case pendingProtocolUpgrade
+    
+    /// The protocol upgrade failure. The recommended action is to retry the status fetch
+    /// operation, or remove the activation locally.
+    case protocolUpgrade(reason: Error?)
+    
+    /// WatchConnectivity feature failure. Check the underlying error or debug log for more details.
+    case watchConnectivity(reason: Error?)
+    
+    
+    // Other errors
+    
+    /// The requested operation was canceled by PowerAuth SDK. This kind of error may occur in situations, when SDK
+    /// needs to cancel an asynchronous operation, but the cancel is not initiated by the application
+    /// itself. For example, if you reset the state of `PowerAuth` during the pending
+    /// fetch for activation status, then the application gets this error in result.
+    case operationCanceled
+    
+    /// The operation failed with an unexpected error.
+    case unexpectedFailure(reason: Error?)
+    
+    /// An internal error or internal data inconsistency in PowerAuth SDK.
+    case internalError(reason: String, underlyingError: Error? = nil)
+}
+
+public extension PowerAuthError {
+
+    /// Returns an underlying error that caused this failure or `nil` if this is the origin of failure.
+    var underlyingError: Error? {
+        switch self {
+            case let .unexpectedFailure(error):
+                return error
+            case let .protocolUpgrade(error):
+                return error
+            case let .watchConnectivity(error):
+                return error
+            case let .internalError(_, underlyingError):
+                return underlyingError
+            default:
+                return nil
+        }
+    }
+}
+
+public extension Error {
+    /// Returns the instance cast as a `PowerAuthError`
+    var asPowerAuthError: PowerAuthError? {
+        self as? PowerAuthError
+    }
+
+    /// Returns the instance cast as a `PowerAuthError`. If casting fails, a `fatalError` with the specified `message` is thrown.
+    func asPowerAuthError(orFailWith message: @autoclosure () -> String, file: StaticString = #file, line: UInt = #line) -> PowerAuthError {
```

Comment:
1. Is it a good idea to have such a helper method (that can cause an uncaught fatal error) public? If the user wants such behavior, he should just cast or use `var asPowerAuthError: PowerAuthError?` above.
2. `#file` is leaking full path to the file, this should be discouraged. Use `#fileID` instead - see "Literal Expression" in https://docs.swift.org/swift-book/ReferenceManual/Expressions.html